pylint/checkers/dataclass_checker.py

```diff
@@ -0,0 +1,137 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+"""Dataclass checkers for Python code."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from astroid import nodes
+
+from pylint.checkers import BaseChecker, utils
+from pylint.interfaces import INFERENCE
+
+if TYPE_CHECKING:
+    from pylint.lint import PyLinter
+
+
+def _is_dataclasses_module(node: nodes.NodeNG) -> bool:
+    """Utility function to check if node is from dataclasses_module."""
+    return isinstance(node, nodes.Module) and node.name == "dataclasses"
+
+
+def _check_name_or_attrname_eq_to(
+    node: nodes.Name | nodes.Attribute, check_with: str
+) -> Any:
+    """Utility function to check either a Name/Attribute node's name/attrname with a
+    given string.
+    """
+    if isinstance(node, nodes.Name):
+        return node.name == check_with
+    return node.attrname == check_with
+
+
+def _is_annotated_with_dataclass(d_nodes: list[nodes.NodeNG]) -> bool:
```

Comment:
You should be able to do most of this with `ClassDef.is_dataclass`. You just need to infer the parents of the node.

Reply:
i realised I can make use of this: https://github.com/PyCQA/astroid/blob/main/astroid/brain/brain_dataclasses.py#L49

do you think it's a good idea?

Reply:
I think it would be better to use `is_dataclass`. We can leave the determining of dataclasses up to `astroid` and keep the logic in one package then.